### ISSUE TITLE
rate limiter delayed for too long

### ISSUE THREAD

**Issue by mayfield** [SauceLLC/sauce4strava#227]
I tried to setup Sauce on a new PC today and my sync is delayed till January 25:

```
18:28:09 DEBUG RateLimiter [streams-min]: period: 65s, usage: 33/30
18:28:09 INFO Rate limited for 41494 minutes
18:28:05 INFO Starting sync job: <Tobi 🐊⚡ Scheller (6500402)>
```

What am I doing wrong?

_Originally posted by @tobisnet in https://github.com/SauceLLC/sauce4strava/discussions/193#discussioncomment-11476329_

**Comment by mayfield** [comment]
Hi @tobisnet.  I created a new issue for this.  Do you have more than one computer syncing using the same google account for chrome?  And if so do both computers have the clocks setup properly?  I.e. is one of them reading a clock that is significantly different than the other?

**Comment by tobisnet** [comment]
To be honest I have it currently on 3 computers in the same private network.
Is the usage limited to a single Chrome instance (one PC)?
One PC had some hard crash a while ago, could be that the time was not in sync with the others for a short while...

**Comment by mayfield** [comment]
Strava limits based on your logged in account.  Sauce tries to stay within the limits and if you have chrome setup to sync settings using your google account the rate limiter code will try to sync as well.  So there isn't anything intrinsically wrong with having 3 devices running sauce  performance.  Are you able to see if the clocks on all 3 computers are the same now?

**Comment by tobisnet** [comment]
I need to check, but assume it, because all computers take internet time automatically. 
Is there something I can "reset" locally?

**Comment by tobisnet** [comment]
I solved it for me (at least) What did I do?
I did import the data in MS Edge and started the sync there, wich finished successfully.
After that exported the data from Edge and imported in Chrome.
Finally the Sync completed in Chrome as well.

**Comment by tobisnet** [comment]
Ahh, still not solved ... new activities run into the same issue :( :

`
19:46:34 DEBUG RateLimiter [streams-min]: period: 65s, usage: 33/30
19:46:34 INFO Rate limited for 37095 minutes
19:46:31 INFO Adding 3 new activities
19:46:28 INFO Starting sync job: <Tobi 🐊⚡ Scheller (6500402)>
`

**Comment by mayfield** [comment]
During the period of the reemergence do you think all the computers you had actively syncing had the same clocks?  Even with internet time, and especially with Windows (if you use it) clocks can get wildly out of sync during suspend/hibernation events.  I'm wondering if this is exacerbating some inherit bug in my design (although I can't see it through just code review yet).

**Comment by mayfield** [comment]
And to unwedge you, these command entered in the Dev tools console (F12) will likely help (at least for a while).

```
sauce.storage.remove('hist-rate-limiter-streams-day', {sync: true});
sauce.storage.remove('hist-rate-limiter-streams-hour', {sync: true});
sauce.storage.remove('hist-rate-limiter-streams-min', {sync: true});
```

**Comment by tobisnet** [comment]
> During the period of the reemergence do you think all the computers you had actively syncing had the same clocks? Even with internet time, and especially with Windows (if you use it) clocks can get wildly out of sync during suspend/hibernation events. I'm wondering if this is exacerbating some inherit bug in my design (although I can't see it through just code review yet).

yes this is possible, had a crash on one PC with BIOS reset, where the PC time was ~1970 for a while ...

**Comment by mayfield** [comment]
I rolled out a change to not use the sync'd storage subsytem that will hopefully resolve this.  Please re-open if you see with versions at or greater than 8.7.11.1.
